azavea/echo-locator
# Feature/kak/add images descriptions#88

## Overview

Adds an input CSV with descriptive text and additional links, including links to images. Query for image metadata for attribution and download thumbnail copies of the images as part of the data processing scripting for images from Wikipedia, which is the source for ~80% of the images. Display added images, text, and links in the neighborhood summary card and detail views.

[Wikipedia image metadata API documentation](https://www.mediawiki.org/wiki/API:Imageinfo).


### Demo

Images in summary cards:
![image](https://user-images.githubusercontent.com/960264/55739985-322bbf00-59f8-11e9-9813-1577e7371160.png)

Details:
![image](https://user-images.githubusercontent.com/960264/55740012-4079db00-59f8-11e9-9b94-e6eb464149e7.png)


### Notes

Non-Wikipedia sourced images are not used here, due to lack of attribution and/or license information. At least some are licensed for reuse, but require attribution. Also, many of the image links are not links to image files, but links to pages for an image.

The downloaded thumbnails fetched by the processing script are not yet in use; instead, the link to the thumbnail is used as a hotlink. Using the downloaded copies would be preferable to prevent broken links. Doing so depends on fixing #112, adding the images to the deployment process, and adding the images to the local live server hosted files.


## Testing Instructions

 * `cd neighborhood_data && ./update_data.sh` to run data processing script
 * Should have thumbnails in `neighborhood_data/images`
 * Should have `image_status.csv` with an explanatory row for each image not found 
 * Should have added source and attribution fields in `neighborhoods.json` GeoJSON file
 * Navigate to map page
 * Should have no remaining placeholders in neighborhood summary cards or detail views
 * Clicking an image should open attribution link in new tab
 * Clicking Wikipedia or official site links should open expected page in new tab


Closes #88
